CONTRIBUTING.md

```diff
@@ -296,6 +296,43 @@ by pushing it to any branch on the main SourceCred repository. This will
 restart the full test suite, which will now actually run. Once your
 tests pass and the PR is approved, we’ll delete the extra branch.
 
-## After your pull request is merged
+## Merging your Pull Request
+
+If you're a new contributor, one of the repository maintainers will merge your pull requests
+for you, once they've been approved by the team. After you've become a seasoned contributor, you can
+ask a maintainer to give you merge permissions.
+
+Once you have merge permissions, the expectations are:
+
+1. Only merge your own pull requests (until you become a maintainer yourself)
+2. Ensure that we have a clean Git history. Every commit in our history should
+   leave the codebase in a functional state, with all tests passing. (This
+   ensures that we can use `git bisect` to hunt down regressions.)
+
+In general, you can ensure that the git history is clean by only using "squash
+and merge", and only merging pull requests that pass CI. "Rebase and merge"
```

Comment:
It might be helpful to have an example of how to do the "squash and merge" (e.g. screenshot of the GitHub button, CLI examples, etc)

Reply:
The examples @BrianBelhumeur descibes could be a collapsible section: _Why "Squash and Merge"_? 

It might be worth going into detail on merging "chained" PRs that are sequentially based on one another. The workflow of interactively rebasing and removing the merged commit is clear once you get the hang of it, but definitely has the potential to trip people up.